app/main.py

```diff
@@ -1,2 +1,88 @@
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Union
+
+
+@dataclass
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.nodes = [[], [], [], [], [], [], [], []]
+
+    def resize(self) -> None:
+        temp = deepcopy(self.nodes)
+        size = len(self.nodes)
+        self.nodes = []
+        for _ in range(2 * size):
+            self.nodes.append([])
+        for node in temp:
+            for item in node:
+                if item:
+                    self.__write(item[0], item[1])
+
+    def __setitem__(self,
+                    key: Union[int, float, bool, str, tuple],
+                    value: Any) -> None:
+        if self.nodes.count([]) <= 1 / 3 * len(self.nodes):
+            self.resize()
+        self.__write(key, value)
+
+    def __write(self,
+                key: Union[int, float, bool, str, tuple],
+                value: Any) -> None:
+        write = True
+        for item in self.nodes[hash(key) % len(self.nodes)]:
+            if key in item:
+                item[1] = value
+                write = False
+                break
+        if write:
+            self.nodes[hash(key) % len(self.nodes)].append([key, value])
```

Comment:
what about colision when you have edged situation: hash(key) % len(self.nodes) == len(self.nodes) - 1 and this node is fufiled

Reply:
"and this node is fufiled" 
I didn't understand how it can be. Exactly how is node fufiled?